opennms-alarms/daemon/src/main/java/org/opennms/netmgt/alarmd/drools/DefaultAlarmService.java

```diff
@@ -81,6 +81,12 @@ public void deleteAlarm(OnmsAlarm alarm) {
             LOG.warn("Alarm disappeared: {}. Skipping delete.", alarm);
             return;
         }
+        // If alarm was in Situation, fire notifications for the Situation
+        if (alarmInTrans.isPartOfSituation()) {
+            for (OnmsAlarm situation : alarmInTrans.getRelatedSituations()) {
+                alarmEntityNotifier.didUpdateRelatedAlarms(situation, situation.getRelatedAlarms());
```

Comment:
This should probably be issued after the call to `alarmDao#delete`. Won't `situation.getRelatedAlarms()` still contain the given alarm we deleted.

Reply:
This is desired behaviour.
Second param of didUpdateRelatedAlarms is 'previousRelatedAlarms'

Reply:
To help clarify the expected behavior - let's assume that originally `situation.getRelatedAlarms()` returns {A,B,C}.

If we're deleting B, then we would expect `didUpdateRelatedAlarms` to be called. When called:
1. `situation.getRelatedAlarms()` should return {A,C} and
1. previousRelatedAlarms should be set to {A,B,C}

It looks like this second part is true, but since the alarm isn't deleted until after the callback is made, calling `situation.getRelatedAlarms()` still returns {A,B,C}.